_integration-test/run.py

```diff
@@ -0,0 +1,197 @@
+import subprocess
+import os
+from functools import lru_cache
+from bs4 import BeautifulSoup
+import httpx
+import pytest
+import sentry_sdk
+import time
+import json
+import re
+
+SENTRY_CONFIG_PY = "sentry/sentry.conf.py"
+SENTRY_TEST_HOST = os.getenv("SENTRY_TEST_HOST", "http://localhost:9000")
+TEST_USER = "test@example.com"
+TEST_PASS = "test123TEST"
+TIMEOUT_SECONDS = 60
+
+
+def poll_for_response(request: str, client: httpx.Client) -> httpx.Response:
+    for i in range(TIMEOUT_SECONDS):
+        response = client.get(
+            request, follow_redirects=True, headers={"Referer": SENTRY_TEST_HOST}
+        )
+        if response.status_code == 200:
+            break
+        time.sleep(1)
+    else:
+        raise AssertionError("timeout waiting for response status code 200")
+    return response
+
+
+@lru_cache
+def get_sentry_dsn(client: httpx.Client) -> str:
+    response = poll_for_response(
+        f"{SENTRY_TEST_HOST}/api/0/projects/sentry/internal/keys/",
+        client,
+    )
+    sentry_dsn = json.loads(response.text)[0]["dsn"]["public"]
+    assert len(sentry_dsn) > 0
+    return sentry_dsn
+
+
+@pytest.fixture(scope="session", autouse=True)
+def configure_self_hosted_environment():
+    response = None
+    subprocess.run(["docker", "compose", "--ansi", "never", "up", "-d"], check=True)
+    for i in range(TIMEOUT_SECONDS):
+        try:
+            response = httpx.get(SENTRY_TEST_HOST, follow_redirects=True)
+        except httpx.ConnectionError:
+            time.sleep(1)
+        if response is not None and response.status_code == 200:
+            break
+    else:
+        raise AssertionError("timeout waiting for self-hosted to come up")
+
+    # Create test user
+    subprocess.run(
+        [
+            "docker",
+            "compose",
+            "exec",
+            "web",
+            "sentry",
+            "createuser",
+            "--force-update",
+            "--superuser",
+            "--email",
+            TEST_USER,
+            "--password",
+            TEST_PASS,
+        ],
+        check=True,
+        text=True,
+        input="y",
```

Comment:
this is what sentry does to avoid needing to pass input: https://github.com/getsentry/sentry/blob/d84a4ea18bae97b933393261a7f1554bb5537e58/scripts/lib.sh#L198